paper.tex

```diff
@@ -78,13 +78,13 @@ \subsubsection{Registrar}
 Registrar is responsible for creating a Registration Contract, which maintains a lookup table of participants, and adding new data providers, consumers and brokers to the decentralized data marketplace.
 
 \subsubsection{Data Provider}
-Data providers, who generate and preserve IoT data, are willing to sell sensing data to consumers. They decide the subscription period, subscription price, data type, sampling frequency. Afterward, they can launch the product on the decentralized data marketplace. Next, they collect raw data with their sensors and send to the broker who assists in uploading data to the network. An MAM channel\cite{MAM} is used to store encrypted data stream.
+Data providers, who generate and preserve IoT data, are willing to sell sensing data to consumers. They decide the subscription period, subscription price, data type, sampling frequency. Afterward, they can launch the product on the decentralized data marketplace. Next, they collect raw data with their sensors and send to the broker who assists in uploading data to the network. An MAM channel\cite{MAM} is used to store encrypted data stream. They receive subscription fees from consumers, which can be used to improve the quantity and accuracy of their sensors.
```

Comment:
Shall we mention subscription in earlier paragraphs?

Reply:
Is it appropriate to refer 'subscription' in the outline of System Architecture?

Reply:
How about unifying occurrence of "IoT data" into "streaming data? We can clarify and define the usage in advance.